src/main/java/com/github/distributedtracing/ContinuationLocal.java

```diff
@@ -0,0 +1,170 @@
+package com.github.distributedtracing;
+
+public final class ContinuationLocal<T> {
+    private static InheritableThreadLocal<Object[]> localCtx =
+            new InheritableThreadLocal<Object[]>();
+    private static volatile int size = 0;
+
+    /**
+     * Return a snapshot of the current ContinuationLocal state.
+     */
+    public static Object[] save() {
```

Comment:
Just a thought: I think it would be easier to locally store a copy of the
work in progress impl in the intended codebases. Whether this is based on
twitter Local or gRPC Context, it is different than most impls.

What if after a round of feedback, we actually integrate a local beta copy
of the desired source into zipkin or brave etc? After it proves to work,
then formalize and raise a PR to replace it with a dep (along with the
merits of doing so).

The 'if you build it, they will come' approach might work, but proving this
out in practice first wouldn't hurt, and probably reduce thrash. As an
increment step, impls at least have a similar approach to start with and
that can help identify gaps that the common one might need, but are
difficult to guess.